sdk/iot/hub/samples/src/paho_iot_hub_c2d_example.c

```diff
@@ -178,21 +182,25 @@ static int connect_device()
 
   MQTTClient_SSLOptions mqtt_ssl_options = MQTTClient_SSLOptions_initializer;
   MQTTClient_connectOptions mqtt_connect_options = MQTTClient_connectOptions_initializer;
+
+  // NOTE: We recommend setting clean session to false in order to receive any pending messages
   mqtt_connect_options.cleansession = false;
   mqtt_connect_options.keepAliveInterval = AZ_IOT_DEFAULT_MQTT_CONNECT_KEEPALIVE_SECONDS;
 
-  size_t username_length;
+  // Get the MQTT user name used to connect to IoT Hub
   if (az_failed(
-          rc = az_iot_hub_client_get_user_name(
-              &client, mqtt_username, sizeof(mqtt_username), &username_length)))
+          rc
+          = az_iot_hub_client_get_user_name(&client, mqtt_username, sizeof(mqtt_username), NULL)))
   {
     printf("Failed to get MQTT clientId, return code %d\n", rc);
     return rc;
   }
 
+  // This sample uses X509 authentication so the password field is set to NULL
   mqtt_connect_options.username = mqtt_username;
   mqtt_connect_options.password = NULL;
 
+  // Set the device cert for authentication
```

Comment:
// Set the device cert for tls mutual authentication

Reply:
FYI, @ericwol-msft, when you have suggestions like this, consider using ` ```suggestion`. It makes it quite easy for the author to accept your feedback, directly on GitHub.

The file is outdated now, but it would look something like this:
```suggestion
// Set the device cert for tls mutual authentication
```

https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request
https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes